infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/server/RouteServerExposer.java

```diff
@@ -124,6 +128,10 @@ public void expose(
             .withTargetPort(servicePort.getName())
             .withServers(externalServers)
             .withLabels(labels)
+            .withHost(
+                domainSuffix != null
+                    ? NameGenerator.generate("route", "." + domainSuffix, 10)
```

Comment:
Since Route host has length limitation, 6 random characters might be enough, to save some length here.

Reply:
done